app/src/ui/merge-branch/merge.tsx

```diff
@@ -143,7 +149,7 @@ export class Merge extends React.Component<IMergeProps, IMergeState> {
               Merge into <strong>{currentBranch ? currentBranch.name : ''}</strong>
             </Button>
           </ButtonGroup>
-          {mergeInfo}
+          {this.renderMergeInfo(invalidBranch)}
```

Comment:
If we go this route, I'd prefer we don't call `renderMergeInfo` at all since we know in advanced if the branch is invalid from line 125.

Something like `invalidBranch ? null : this.renderMergeInfo` would do; however, it deviates from the pattern we use in most other components. Here's an [example](https://github.com/desktop/desktop/blob/master/app/src/ui/add-repository/create-repository.tsx#L367-L369).